### PR TITLE
Add a new Beer

### PR DESCRIPTION
**Pull request by arvindsouza**

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -36,3 +35,25 @@ export function updateBeer(id, rating) {
...
+
+export function addNewBeer(data, file) {
+  const request = db
+    .collection('Beers')
+    .doc()
+    .set(data)
+    .then(() => {
+      return new Promise(resolve => {
+        return resolve('Success');
+      });
```
Comment (lkostrowski) [inline src/actions/index.js]: You can do `return Promise.resolve('Success')` or even shorter:
`.then( () => 'Success' )`

File: src/actions/index.js [hunk 2]
```diff
@@ -36,3 +35,25 @@ export function updateBeer(id, rating) {
...
+    });
+
+  if (data.img) {
+    var refA = storage.child(data.img);
```
Comment (lkostrowski) [inline src/actions/index.js]: dont use vars. Add to eslint proper rule

File: src/containers/addNew.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        rating: false,
+        desc: false,
+      },
+      beerName: '',
+      rating: 1,
+      desc: '',
+      file: '',
+      isDisabled: true,
+    };
+  }
```
Comment (lkostrowski) [inline src/containers/addNew.js]: If you move state to class property (outside constructor), you dont have to add constructor function at all

File: src/containers/addNew.js [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+          this.setState({
+            isDisabled: false,
+          });
+        } else
+          this.setState({
+            isDisabled: true,
+          });
+      },
+    );
+  };
```
Comment (lkostrowski) [inline src/containers/addNew.js:51]: Lines 38-45 should be some other function called here (checkDisabledFields or soemthing like that)

File: src/containers/addNew.js [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+  renderNumberField = field => {
+    return (
+      <div className="form-group" id="rating-field">
+        <label className="field-label">Enter the rating</label>
+        <label className="stars">
+          <Ratings
+            rating={this.state.rating}
+            widgetEmptyColors="rgb(255,239,212)"
+            widgetHoverColors="rgb(173, 21, 21)"
```
Comment (lkostrowski) [inline src/containers/addNew.js]: move color variables to sobie variables/theme/config/top of the file, dont store them inline

File: src/containers/addNew.js [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+            changeRating={this.changeTheRating}
+            widgetDimensions='30px'
```
Comment (lkostrowski) [inline src/containers/addNew.js]: This also should be moved to some variable

File: src/containers/addNew.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+            <Ratings.Widget />
+            <Ratings.Widget />
+          </Ratings>
+        </label>
+        <input
+          type="number"
+          className="form-control rating-input"
+          onChange={this.handleChange(field)}
+          value={this.state.rating}
+          readOnly
```
Comment (lkostrowski) [inline src/containers/addNew.js:109]: why readOnly?

File: src/containers/addNew.js [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+      beerName: e.target[0].value,
+      rating: e.target[1].value,
+      desc: e.target[2].value,
+      img: temp,
+    };
+
+    this.props.addNewBeer(values, file).then(() => {
+      this.props.history.push('/beers');
+    });
+  };
```
Comment (lkostrowski) [inline src/containers/addNew.js:152]: This function is totally unreadable. Why do you take values from event if you have them in state?

**Review by lkostrowski — CHANGES_REQUESTED** [review]
I recommend to use great library Formik, to handle form management for you 

https://jaredpalmer.com/formik/docs/overview

File: src/containers/addNew.js (hunk 7 above)
Comment (arvindsouza) [reply src/containers/addNew.js:109]: the values are being obtained through the <Ratings> tag which shows a set of 5 stars, this input is just to display the numerical value of the rating.

File: src/containers/addNew.js (hunk 8 above)
Comment (arvindsouza) [reply src/containers/addNew.js:152]: I just didn't think to get them from state, I'll streamline it.

File: src/containers/addNew.js [hunk 9]
```diff
@@ -7,42 +7,45 @@ import classNames from 'classnames';
...
+  disableButton = (beerName, desc) => {
+    if (beerName !== '' && desc !== '') {
+      this.setState({
+        isDisabled: false,
+      });
+    } else
+      this.setState({
+        isDisabled: true,
+      });
+  };
```
Comment (lkostrowski) [inline src/containers/addNew.js:40]: This function says "disable button", but it takes two atributes nad sometimes disables and sometimes enables it.
Thats wrong. Name should match what functions does

**Review by lkostrowski — CHANGES_REQUESTED** [review]

**Review by lkostrowski — APPROVED** [review]
This name could be better but leave it for now. Maybe it will be resovled while you implement Formik

**Pull request merged**